src/tools/rbd_mirror/GroupReplayer.cc

```diff
@@ -561,37 +580,50 @@ void GroupReplayer<I>::handle_bootstrap_group(int r) {
     reregister_admin_socket_hook();
   }
 
+//FIXME : Relook at this once the Bootstrap no longer reuses values.
+// Since a rerun may end up with variables populated by a previous run,
+// it is safer to destroy the image_replayers now.
+  m_destroy_replayers = true;
   if (finish_start_if_interrupted()) {
     return;
   } else if (r == -ENOENT) {
     finish_start_fail(r, "group removed");
     return;
   } else if (r == -EREMOTEIO) {
+    m_destroy_replayers = true;
     finish_start_fail(r, "remote group is non-primary");
     return;
   } else if (r == -EEXIST) {
     finish_start_fail(r, "split-brain detected");
     return;
+  } else if (m_remote_group_id.empty()){
```

Comment:
Until we have bootstrap fixes, we better defend for primary here:

```
} else if (!m_local_group_ctx.primary && m_remote_group_id.empty()){
    finish_start_fail(-EINVAL, "remote is not ready yet");
    return;
}  ...
```

Reply:
What scenario is this fixing?